nanofi/include/core/cstructs.h

```diff
@@ -76,6 +80,12 @@ typedef struct {
  * ##################################################################
  */
 
+typedef struct{
+  char *key;
```

Comment:
Can we have this as const char *?
A bit better user experience if you can write something like 
```
key_value.key = "some string" 
```

Reply:
The example code I took this from had this as char*, not sure we can do this. I'd have to look at the library.

Reply:
I'll add that I'm not against this, but the code pointer's management was a little different. Again, I'll take a look at if this makes sense. core will be things that don't reflect a user experience though. That is code that's not part of our public API.